communication/mavsdkvehicleserver.cpp

```diff
@@ -498,14 +501,20 @@ void MavsdkVehicleServer::setWaypointFollower(QSharedPointer<WaypointFollower> w
     connect(this, &MavsdkVehicleServer::pauseWaypointFollower, mWaypointFollower.get(), &WaypointFollower::stop);
     connect(this, &MavsdkVehicleServer::resetWaypointFollower, mWaypointFollower.get(), &WaypointFollower::resetState);
     connect(this, &MavsdkVehicleServer::clearRouteOnWaypointFollower, mWaypointFollower.get(), &WaypointFollower::clearRoute);
-    connect(this, &MavsdkVehicleServer::startFollowPoint, mWaypointFollower.get(), &WaypointFollower::startFollowPoint);
 }
 
 void MavsdkVehicleServer::setMovementController(QSharedPointer<MovementController> movementController)
 {
     mMovementController = movementController;
 }
 
+void MavsdkVehicleServer::setFollowPoint(QSharedPointer<FollowPoint> followPoint)
+{
+    mFollowPoint = followPoint;
+    connect(this, &MavsdkVehicleServer::startFollowPoint, mFollowPoint.get(), &FollowPoint::startFollowPoint);
```

Comment:
Related to the previous comment: previously, WPF and FP had a  single STM that made sure that both autopilots could not be active at the same time. How is this now ensured? Is the change above sufficient or will MavsdkVehicleServer::startFollowPoint need to call sth like (and vice verse for WPF)
```
if (!mWaypointFollower.isNull() && mWaypointFollower->isActive())
                emit pauseWaypointFollower();
```

Reply:
The way to change mode is through the above function, which with the suggested change, will halt the other follower. I think it should be sufficient for now. This check shouldn't need to be performed for each communication interface, so a common STM is a better solution...